Userland/Libraries/LibWeb/FileAPI/Blob.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2022, Kenneth Myhra <kennethmyhra@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/StdLibExtras.h>
+#include <LibJS/Runtime/ArrayBuffer.h>
+#include <LibWeb/FileAPI/Blob.h>
+
+namespace Web::FileAPI {
+
+// https://w3c.github.io/FileAPI/#ref-for-dom-blob-blob
+Blob::Blob(Optional<Vector<BlobPart>> const& blob_parts, Optional<BlobPropertyBag> const& options)
+{
+    // 1. If invoked with zero parameters, return a new Blob object consisting of 0 bytes, with size set to 0, and with type set to the empty string.
+    if (!blob_parts.has_value() && !options.has_value())
+        return;
+
+    // 2. Let bytes be the result of processing blob parts given blobParts and options.
+    if (blob_parts.has_value())
+        m_byte_buffer = process_blob_parts(blob_parts.value());
+
+    if (options.has_value())
+        m_type = options->type;
+}
+
+Blob::Blob(ByteBuffer const& byte_buffer, String const& type)
+    : m_byte_buffer(move(byte_buffer))
+    , m_type(move(type))
+{
+}
+
+// https://w3c.github.io/FileAPI/#process-blob-parts
+ByteBuffer Blob::process_blob_parts(Vector<BlobPart> const& blob_parts)
+{
+    // 1. Let bytes be an empty sequence of bytes.
+    ByteBuffer bytes {};
+
+    // 2. For each element in parts:
+    for (auto const& blob_part : blob_parts) {
+        // 1. If element is a USVString, run the following substeps:
+        if (blob_part.has<String>()) {
+            // 1. Let s be element.
+            auto s = blob_part.get<String>();
```

Comment:
You probably don't want to copy this, `auto& s = ...`
Also nitpick (feel free to ignore and all that), can this variable have a nicer name? `element` or `string` or something?